src/ppx_import.ml

```diff
@@ -369,52 +369,48 @@ let type_declaration ~tool_name ~input_name (type_decl : Ppxlib.type_declaration
   | { ptype_attributes
     ; ptype_name
     ; ptype_manifest =
-        Some {ptyp_desc = Ptyp_extension ({txt = "import"; loc}, payload); _}
-    ; _ } -> (
-    match payload with
-    | PTyp ({ptyp_desc = Ptyp_constr ({txt = lid; loc}, _); _} as manifest) ->
-      if tool_name = "ocamldep" then
-        (* Just put it as manifest *)
-        if is_self_reference ~input_name lid then
-          {type_decl with ptype_manifest = None}
-        else {type_decl with ptype_manifest = Some manifest}
-      else
-        Ast_helper.with_default_loc loc (fun () ->
-            let ttype_decl =
-              let env = Lazy.force lazy_env in
-              match lid with
-              | Lapply _ ->
-                Location.raise_errorf ~loc
-                  "[%%import] cannot import a functor application %s"
-                  (string_of_lid lid)
-              | Lident _ as head_id ->
-                (* In this case, we know for sure that the user intends this lident
-                   as a type name, so we use Typetexp.find_type and let the failure
-                   cases propagate to the user. *)
-                Compat.find_type env ~loc head_id |> snd
-              | Ldot (parent_id, elem) ->
-                let sig_items = locate_sig ~loc env parent_id in
-                get_type_decl ~loc sig_items parent_id elem
-            in
-            let m, s =
-              if is_self_reference ~input_name lid then (None, [])
-              else
-                let subst = subst_of_manifest manifest in
-                let subst =
-                  subst
-                  @ [ ( `Lid (Lident (Longident.last_exn lid))
-                      , Ast_helper.Typ.constr
-                          {txt = Lident ptype_name.txt; loc = ptype_name.loc}
-                          [] ) ]
-                in
-                (Some manifest, subst)
-            in
-            let ptype_decl =
-              ptype_decl_of_ttype_decl ~manifest:m ~subst:s ptype_name
-                ttype_decl
-            in
-            {ptype_decl with ptype_attributes} )
-    | _ -> Location.raise_errorf ~loc "Invalid [%%import] syntax" )
+        Some ({ptyp_desc = Ptyp_constr ({txt = lid; loc}, _); _} as manifest)
+    ; _ } ->
+    if tool_name = "ocamldep" then
+      (* Just put it as manifest *)
+      if is_self_reference ~input_name lid then
+        {type_decl with ptype_manifest = None}
+      else {type_decl with ptype_manifest = Some manifest}
+    else
+      Ast_helper.with_default_loc loc (fun () ->
+          let ttype_decl =
+            let env = Lazy.force lazy_env in
+            match lid with
+            | Lapply _ ->
+              Location.raise_errorf ~loc
+                "[%%import] cannot import a functor application %s"
+                (string_of_lid lid)
```

Comment:
Again, this comment is not really related to the PR, but it is a bad practice to raise errors with ppxlib, as it prevents other rewriters to work and more importantly it prevents merlin to work...
Instead, errors should be embedded. I have written a section in the ppxlib manual about his [here](https://ocaml-ppx.github.io/ppxlib/ppxlib/manual.html#handling-errors), which includes a guide for [migrating from raising to embedding](https://ocaml-ppx.github.io/ppxlib/ppxlib/manual.html#migrating-from-raising-to-embedding-errors).

I understand that this should most certainly be part of another PR!